transfer-eth-erc20/from-near.js

```diff
@@ -173,49 +173,27 @@ class TransferEthERC20FromNear {
   }) {
     // Wait for the block with the given receipt/transaction in Near2EthClient.
     try {
-      const outcomeBlockHeight = new BN(outcomeBlock.header.height)
-      let clientBlock
+      const outcomeBlockHeight = Number(outcomeBlock.header.height)
       let clientBlockHeight
-      let clientBlockValidAfter
-      let clientBlockHashB58
-      let clientBlockHashHex
+      let clientBlockHash
       while (true) {
-        clientBlock = await clientContract.methods.head().call()
-        clientBlockHeight = new BN(clientBlock.height)
-        clientBlockValidAfter = new BN(clientBlock.validAfter)
-        clientBlockHashHex = await clientContract.methods
-          .blockHashes(clientBlockHeight)
-          .call()
-        clientBlockHashB58 = bs58.encode(toBuffer(clientBlockHashHex))
-        console.log(
-          `Current light client head is: hash=${clientBlockHashB58}, height=${clientBlockHeight.toString()}`
-        )
+        let clientState = await clientContract.methods.bridgeState().call()
+        clientBlockHeight = Number(clientState.currentHeight)
+        let clientBlockValidAfter = Number(clientState.nextValidAt)
+        clientBlockHash = bs58.encode(toBuffer(await clientContract.methods.blockHashes(clientBlockHeight).call()))
+
+        console.log(`Current light client head is: hash=${clientBlockHash}, height=${clientBlockHeight}`)
 
-        const chainBlock = await robustWeb3.getBlock('latest')
-        const chainBlockTimestamp = new BN(chainBlock.timestamp)
-        if (clientBlockHeight.gt(outcomeBlockHeight)) {
-          console.log(
-            `Near2EthClient block is at ${clientBlockHeight.toString()} which is further than the needed block ${outcomeBlockHeight.toString()}`
-          )
+        if (clientBlockHeight > outcomeBlockHeight) {
```

Comment:
Does JS int okay here? it's up to 2^52, bigger than that will become imprecise

Reply:
I think that we won't reach block height of more than 2<sup>53</sup>≅9·10<sup>15</sup> any time soon.